src/main/scala/com/criteo/hive/HiveClientLogger.scala

```diff
@@ -0,0 +1,13 @@
+package com.criteo.hive
+
+/**
+  * Abstraction for hive client logging.
+  */
+trait HiveClientLogger {
+
+  def info(str: CharSequence)
+
+  def error(str: CharSequence)
+
+  def debug(str: CharSequence)
```

Comment:
Why CharSequence instead of String?  Also, you probably want to declare these as by-name parameters for lazy evaluation, especially in the event of debugging calls (which may be expensive if evaluated).  Also, declare return types, even if it's Unit, in traits, eg:
```
trait HiveClientLogger {
  def info(line: => String): Unit

  def error(line: => String): Unit

  def debug(line: => String): Unit
}
```

See: https://docs.scala-lang.org/tour/by-name-parameters.html